core/src/env/test_env.rs

```diff
@@ -567,6 +609,35 @@ where
     fn dispatch_raw_call(data: &[u8]) {
         TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().dispatch_call(data))
     }
+
+    fn call_invoke(
+        callee: T::AccountId,
+        gas: u64,
+        value: T::Balance,
+        input_data: &[u8],
+    ) -> Result<(), CallError> {
+        let callee = &(callee.encode())[..];
+        let value = &(value.encode())[..];
+        let _return_data = TEST_ENV_DATA
+            .with(|test_env| test_env.borrow_mut().call(callee, gas, value, input_data));
+        Ok(())
+    }
+
+    fn call_evaluate<U: Decode>(
+        callee: T::AccountId,
+        gas: u64,
+        value: T::Balance,
+        input_data: &[u8],
+    ) -> Result<U, CallError> {
+        let callee = &(callee.encode())[..];
+        let value = &(value.encode())[..];
+        let return_data = TEST_ENV_DATA
+            .with(|test_env| test_env.borrow_mut().call(callee, gas, value, input_data));
+        match U::decode(&mut &return_data[..]) {
```

Comment:
can be substituted with `.map(|bytes| U::decode(&mut &bytes[..]).map_err(CallError)`

Reply:
Ok, i'll try!